rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMStateCaching.java

```diff
@@ -102,70 +78,44 @@ private static void init_() {
                         String password = PortalConfiguration.RM_CACHE_PASSWORD.getValueAsString();
-                        rm.init(url, new CredData(login, password));
+                        rmProxyActiveObject.init(url, new CredData(login, password));
                     }
                 }
             } catch (Exception e) {
                 logger.warn("Could not connect to resource manager at " + url + " retrying in 8 seconds", e);
-                if (rm != null) {
-                    PAActiveObject.terminateActiveObject(rm, true);
-                    rm = null;
+                if (rmProxyActiveObject != null) {
+                    PAActiveObject.terminateActiveObject(rmProxyActiveObject, true);
+                    rmProxyActiveObject = null;
                 }
                 new Sleeper(8 * 1000, logger).sleep();
                 continue;
             }
         }
     }
 
-    private static void run_() {
-        rmUpdater = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                boolean isDebugEnabled = logger.isDebugEnabled();
-                long startTime = 0;
-
-                while (!kill) {
-                    try {
-                        if (isDebugEnabled) {
-                            startTime = System.currentTimeMillis();
-                        }
-
-                        state = PAFuture.getFutureValue(rm.getRMInitialState());
-
-                        if (isDebugEnabled) {
-                            logger.debug("Updated RM initial state in " + (System.currentTimeMillis() - startTime) +
-                                         "ms");
-                        }
-                    } catch (Throwable t) {
-                        logger.error("Exception occurrend while updating RM state cache, connection reset", t);
-                        init_();
-                    }
-
-                    new Sleeper(refreshInterval, logger).sleep();
-                }
-            }
-        });
-        rmUpdater.setName("RM Initial State Cache Updater");
-        rmUpdater.setDaemon(true);
-        rmUpdater.start();
-    }
-
     /**
-     * @return cached RM State as returned by {@link RMProxyUserInterface#getMonitoring()}
+     * @return cached RM State as returned by {@link RMProxyUserInterface#getRMInitialState(long)}
      */
-    public static RMInitialState getRMInitialState() {
-        return state;
-    }
+    public static RMInitialState getRMInitialState(long counter) {
+        try {
+            long startTime = System.currentTimeMillis();
 
-    /**
-     * Stop the RM State polling thread.
-     */
-    public static void kill() {
-        RMStateCaching.kill = true;
-    }
+            // we request the state of the RM directly via rmProxy object
+            // and not via ActiveObject wrap of it.
+            final RMInitialState state = rmProxy.getRMInitialState(counter);
```

Comment:
Why do we need to pass by the object reference instead of the active object?

Reply:
Fabien suggested to access RMProxyUserInterface directly

Reply:
@fviale I don't agree here. If we make rmProxy an active object we should always use it as it. We could get rid of this rmProxy reference and only keep and use the rmProxyActiveObject reference (in this case we can just rename it rmProxy), because:
1) we only use the plain old object in the above line.
2) while the current thread gets the initial state of the rmProxy object, the thread of the rmProxyActiveObject could manipulate it as well.